serde/src/main/java/com/amazon/ionhiveserde/objectinspectors/utils/IonPrimitiveReader.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.ionhiveserde.objectinspectors.utils;
+
+import com.amazon.ion.IonReader;
+import com.amazon.ion.IonSystem;
+import com.amazon.ion.IonTimestamp;
+import com.amazon.ion.IonType;
+import com.amazon.ion.Timestamp;
+import com.amazon.ion.system.IonSystemBuilder;
+import com.google.common.collect.ImmutableSet;
+
+import java.math.BigDecimal;
+import java.util.Date;
+import java.util.Set;
+
+public class IonPrimitiveReader {
+    private static final IonSystem SYSTEM = IonSystemBuilder.standard().build();
```

Comment:
Might as well create an `IonReaderBuilder` directly; no need for an `IonSystem`.

```
private static final IonReaderBuilder READER_BUILDER = IonReaderBuilder.standard().immutable();
```

Then line 85 is

```
IonReader reader = READER_BUILDER.build(ionPrimitive);
```

Reply:
Ah cool, updated! I used the system since I was running into an NPE when trying to build the reader with:

```
 IonReaderBuilder readerBuilder = IonSystemBuilder.standard().getReaderBuilder();
```

But definitely cleaner to use the reader builder.